### PR TITLE
ccip home

### PR DESCRIPTION
**Pull request by RensR**
Fresh PR

### CODE REVIEW

**Review by makramkd — APPROVED** [review]

File: contracts/scripts/native_solc_compile_all_ccip [hunk 1]
```diff
@@ -73,8 +73,7 @@ compileContract ccip/RMN.sol
 compileContract ccip/ARMProxy.sol
 compileContract ccip/tokenAdminRegistry/TokenAdminRegistry.sol
 compileContract ccip/tokenAdminRegistry/RegistryModuleOwnerCustom.sol
-compileContract ccip/capability/CCIPConfig.sol
-compileContract ccip/capability/interfaces/IOCR3ConfigEncoder.sol
```
Comment (RensR) [inline contracts/scripts/native_solc_compile_all_ccip:77]: merged into ICCIPEncodingUtils

File: contracts/src/v0.8/ccip/capability/CCIPHome.sol [hunk 2]
```diff
@@ -0,0 +1,604 @@
...
+  // ================================================================
+
+  /// @notice Returns the capabilities registry address.
+  /// @return The capabilities registry address.
+  function getCapabilityRegistry() external view returns (address) {
+    return i_capabilitiesRegistry;
+  }
+
+  /// @inheritdoc IERC165
+  function supportsInterface(bytes4 interfaceId) external pure override returns (bool) {
```
Comment (makramkd) [inline contracts/src/v0.8/ccip/capability/CCIPHome.sol]: I think you had a comment on the older CCIPHome PR as to why this is implemented - i.e that its expected by the CapabilitiesRegistry. Worth putting it back here?

**Review by makramkd — APPROVED** [review]

File: contracts/src/v0.8/ccip/FeeQuoter.sol [hunk 3]
```diff
@@ -94,8 +94,8 @@ contract FeeQuoter is AuthorizedCallers, IFeeQuoter, ITypeAndVersion, IReceiver,
     uint32 maxPerMsgGasLimit; //                 │ Maximum gas limit for messages targeting EVMs
     uint32 destGasOverhead; //                   │ Gas charged on top of the gasLimit to cover destination chain costs
     uint16 destGasPerPayloadByte; //             │ Destination chain gas charged for passing each byte of `data` payload to receiver
-    uint32 destDataAvailabilityOverheadGas; //   | Extra data availability gas charged on top of the message, e.g. for OCR
-    uint16 destGasPerDataAvailabilityByte; //    | Amount of gas to charge per byte of message data that needs availability
+    uint32 destDataAvailabilityOverheadGas; //   │ Extra data availability gas charged on top of the message, e.g. for OCR
+    uint16 destGasPerDataAvailabilityByte; //    │ Amount of gas to charge per byte of message data that needs availability
```
Comment (RensR) [inline contracts/src/v0.8/ccip/FeeQuoter.sol:98]: fixed the not full height comment

File: contracts/src/v0.8/ccip/capability/CCIPHome.sol [hunk 4]
```diff
@@ -0,0 +1,605 @@
...
+  /// @notice Returns the active config digest for for a given key.
+  /// @param donId The key of the plugin to get the config digests for.
+  function getActiveDigest(uint32 donId, Internal.OCRPluginType pluginType) public view returns (bytes32) {
+    return s_configs[donId][pluginType][s_activeConfigIndex].configDigest;
+  }
+
+  /// @notice Returns the candidate config digest for for a given key.
+  /// @param donId The key of the plugin to get the config digests for.
+  function getCandidateDigest(uint32 donId, Internal.OCRPluginType pluginType) public view returns (bytes32) {
+    return s_configs[donId][pluginType][s_activeConfigIndex ^ 1].configDigest;
```
Comment (connorwstein) [inline contracts/src/v0.8/ccip/capability/CCIPHome.sol]: can have the same helper as RMNHome getActiveConfigIndex()

**Review by connorwstein — APPROVED** [review]

File: contracts/src/v0.8/ccip/capability/CCIPHome.sol (hunk 4 above)
Comment (RensR) [reply contracts/src/v0.8/ccip/capability/CCIPHome.sol]: Fixed the naming of the RMN version, it was excluded from solhint so the naming didn't trigger an issue. That's fixed now.

**Pull request merged**